src/common/meta/src/peer.rs

```diff
@@ -64,3 +68,56 @@ impl Display for Peer {
         write!(f, "peer-{}({})", self.id, self.addr)
     }
 }
+
+/// can query peer given a node id
+#[async_trait::async_trait]
+pub trait PeerLookupService {
+    async fn datanode(&self, cluster_id: ClusterId, id: DatanodeId) -> Result<Option<Peer>, Error>;
+    async fn flownode(&self, cluster_id: ClusterId, id: FlownodeId) -> Result<Option<Peer>, Error>;
+}
+
+pub type PeerLookupServiceRef = Arc<dyn PeerLookupService + Send + Sync>;
+
+/// A dummy implementation of [PeerLookupService] for testing purpose.
+pub struct DummyPeerLookupService;
+
+#[async_trait::async_trait]
+impl PeerLookupService for DummyPeerLookupService {
+    async fn datanode(
+        &self,
+        _cluster_id: ClusterId,
+        _id: DatanodeId,
+    ) -> Result<Option<Peer>, Error> {
+        Ok(None)
+    }
+
+    async fn flownode(
+        &self,
+        _cluster_id: ClusterId,
+        _id: FlownodeId,
+    ) -> Result<Option<Peer>, Error> {
+        Ok(None)
+    }
+}
```

Comment:
nit: add a `#[cfg(test)]` gate

Reply:
https://stackoverflow.com/questions/74687548/export-a-type-under-cfgtest-in-crate-a-so-that-it-can-be-used-for-unit-test
can't do that because `meta-srv` crate use this 

> nit: add a `#[cfg(test)]` gate